drivers/sdhc/sdhc_cdns.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT cdns_sdhc
+
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/reset.h>
+
+#include "sdhc_cdns_ll.h"
+
+#define SDHC_CDNS_DESC_SIZE	(1<<20)
+#define COMBOPHY_ADDR_MASK	0x0000FFFFU
+
+#define DEV_CFG(_dev)	((const struct sdhc_cdns_config *)(_dev)->config)
+#define DEV_DATA(_dev)	((struct sdhc_cdns_data *const)(_dev)->data)
+
+LOG_MODULE_REGISTER(sdhc_cdns, CONFIG_SDHC_LOG_LEVEL);
+
+/* SDMMC operations FPs are the element of structure*/
+static const struct sdhc_cdns_ops *cdns_sdmmc_ops;
+
+struct sdhc_cdns_config {
+	DEVICE_MMIO_NAMED_ROM(reg_base);
+	DEVICE_MMIO_NAMED_ROM(combo_phy);
+	/* Clock rate for host */
+	uint32_t clk_rate;
+	/* power delay prop for host */
+	uint32_t power_delay_ms;
+	/* run time device structure */
+	const struct device *cdns_clk_dev;
+	/* type to identify a clock controller sub-system */
+	clock_control_subsys_t clkid;
+	/* Reset controller device configuration. */
+	const struct reset_dt_spec reset_sdmmc;
+	const struct reset_dt_spec reset_sdmmcocp;
+	const struct reset_dt_spec reset_softphy;
+};
+
+struct sdhc_cdns_data {
+	DEVICE_MMIO_NAMED_RAM(reg_base);
+	DEVICE_MMIO_NAMED_RAM(combo_phy);
+	/* Host controller parameters */
+	struct sdhc_cdns_params params;
+	/* sdmmc device informartaion for host */
+	struct sdmmc_device_info info;
+	/* Input/Output configuration */
+	struct sdhc_io host_io;
+};
+
+static int sdhc_cdns_request(const struct device *dev,
+	struct sdhc_command *cmd,
+	struct sdhc_data *data)
+{
+	int ret = 0;
+	struct sdmmc_cmd cdns_sdmmc_cmd;
+
+	if (cmd == NULL) {
+		LOG_ERR("Wrong CMD parameter");
+		return -EINVAL;
+	}
+
+	/* Initialization of command structure */
+	cdns_sdmmc_cmd.cmd_idx =  cmd->opcode;
+	cdns_sdmmc_cmd.cmd_arg = cmd->arg;
+	cdns_sdmmc_cmd.resp_type = (cmd->response_type & SDHC_NATIVE_RESPONSE_MASK);
+
+	/* Sending command as per the data or non data */
+	if (data) {
+		ret = cdns_sdmmc_ops->prepare(data->block_addr, (uintptr_t)data->data,
+				data);
+		if (ret != 0) {
+			LOG_ERR("DMA Prepare failed");
+			return -EINVAL;
+		}
+	}
+		// cdns_sdmmc_cmd.resp_type = MMC_RESPONSE_R1;
+	// } else {
```

Comment:
Two things here:
- these comments will fail CI, Zephyr only allows C89 style `/* */` comments
- Why is this block commented out? It seems like it would be required for the driver to function. Is the implementation of this now in `sdhc_cdns_ll.c`?